### PR TITLE
Remove obsolete code in `Castle.Facilities.Logging`

### LINKED ISSUE #616: Deleting obsolete `Castle.Facilities.Logging` code, handling `FacilityConfig`
While looking at adding `net6.0` as a target, I took a look at `Castle.Facilities.Logging` (there is a reference to `DiagnosticsLoggerFactory` which theoretically can work under `net6.0-win` but not with the way the code is currently structured). I noticed how much is `obsolete` in this project, since 2017, so took a look at removing code.

I'm struggling with how to handle `ReadLoggingApi`:
https://github.com/castleproject/Windsor/blob/5a32d4f08f126d6b70191393e099c5e4a0031423/src/Castle.Facilities.Logging/LoggingFacility.cs#L399-L410

The `LoggerImplementation` enum is marked obsolete, so looking at `FacilityConfig.Attributes["loggingApi"]` for the type to create is made more complicated. Any thoughts on how to handle this? Is configuration via `FacilityConfig` common?

### PR DESCRIPTION
**Pull request by Jevonius**
This is the work to resolve #616; it's built off the #630 branch, so I anticipate rebasing this branch once that's merged down. There will be more noise in this PR until then.

### CODE REVIEW

**Review by jonorossi — APPROVED** [review]
Holy crap, YES YES YES!!! I can't believe it was 5 years ago I deprecated these. Really glad to see that code finally go, it caused so much pain during the early .NET Core migration days.

🎉

File: src/Castle.Facilities.AspNet.SystemWeb.Tests/Castle.Facilities.AspNet.SystemWeb.Tests.csproj [hunk 1]
```diff
@@ -19,10 +19,10 @@
 	</PropertyGroup>
 
 	<ItemGroup>
-    <PackageReference Include="NUnit" Version="3.13.3" />
-    <PackageReference Include="NUnit3TestAdapter" Version="4.2.1" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.3.2" />
-		<PackageReference Include="Microsoft.TestPlatform.ObjectModel" Version="17.3.2" />
+		<PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.3.2" />
+		<PackageReference Include="Microsoft.TestPlatform.ObjectModel" Version="11.0.0" />
```
Comment (jonorossi) [inline src/Castle.Facilities.AspNet.SystemWeb.Tests/Castle.Facilities.AspNet.SystemWeb.Tests.csproj:23]: You've downgrade `Microsoft.TestPlatform.ObjectModel` here which is causing the build to fail.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ Bugfixes:
 
 Breaking Changes:
 - Microsoft.Extensions.Hosting related methods have been removed from the Castle.Windsor.Extensions.DependencyInjection package to the Castle.Windsor.Extensions.Hosting package (@ikkentim, #625, #628)
+- Obsolete components in Castle.Facilities.Logging have been removed. Extensions methods for built-in logging factories have been added. (@Jevonius, #616)
```
Comment (jonorossi) [inline CHANGELOG.md:15]: Could you list out the public APIs that have been removed like we do in Castle Core (https://github.com/castleproject/Core/blob/master/CHANGELOG.md#440-2019-04-05) so it makes it clear to upgraders and to enable search. You don't need every enum member, just the enum type name.

File: src/Castle.Facilities.AspNet.SystemWeb.Tests/Castle.Facilities.AspNet.SystemWeb.Tests.csproj (hunk 1 above)
Comment (Jevonius) [reply src/Castle.Facilities.AspNet.SystemWeb.Tests/Castle.Facilities.AspNet.SystemWeb.Tests.csproj:23]: Odd, I remember fighting an issue with this versioning, and needed to downgrade again (had upgraded) to fix it, but can't remember what. Will take a look.

Comment (Jevonius) [reply src/Castle.Facilities.AspNet.SystemWeb.Tests/Castle.Facilities.AspNet.SystemWeb.Tests.csproj:23]: There's a couple of versions in use across the solution - `src/Castle.Facilities.AspNet.Mvc.Tests/Castle.Facilities.AspNet.Mvc.Tests.csproj`, `src/Castle.Facilities.AspNet.WebApi.Tests/Castle.Facilities.AspNet.WebApi.Tests.csproj`, and `src/Castle.Windsor.Tests/Castle.Windsor.Tests.csproj` use `11.0.0`, so correction to above, I suspect this one got changed to match. It's _possible_ that it's to work around a test issue with the older NuGet. The problem is alluded to here: https://github.com/microsoft/vstest/issues/1417 - one of the DLLs, I forget which one, has a design-time reference to this old version, never actually used, except Windsor's assembly loading causes it to try. I'll try and remember/work out which one.

**Comment by Jevonius** [comment]
I've re-based this, and build now passing again. Still need to update the docs regarding deprecated APIs.

File: CHANGELOG.md (hunk 2 above)
Comment (lunchin) [reply CHANGELOG.md:15]: Here are the removed classes and methods.  Is anything else needed to merge this and create a release.  I am happy to help get this out the door.

- Removed class Castle.Facilities.Logging.LoggerImplementation
- Removed constructor Castle.Facilities.Logging.LoggingFacility(LoggerImplementation loggingApi)
- Removed constructor Castle.Facilities.Logging.LoggingFacility(LoggerImplementation loggingApi, string configFile)
- Removed constructor Castle.Facilities.Logging.LoggingFacility(string customLoggerFactory, string configFile)
- Removed method Castle.Facilities.Logging.LoggingFacility.LogUsing(LoggerImplementation loggingApi)
- Removed method Castle.Facilities.Logging.LoggingFacility.UseLog4Net()
- Removed method Castle.Facilities.Logging.LoggingFacility.UseLog4Net(string configFile)
- Removed method Castle.Facilities.Logging.LoggingFacility.UseNLog()
- Removed method Castle.Facilities.Logging.LoggingFacility.UseNLog(string configFile)

**Comment by Regenhardt** [comment]
Looks like this is only missing the changelog update now. @Jevonius would you accept those as a PR into your feature branch? I think @lunchin did the hard work already so I'd happily add that to the changelog so we get get #621 out the door.

**Comment by thibnes** [comment]
@jonorossi is there something preventing us from moving forward here as suggested by @Regenhardt?

File: src/Castle.Facilities.AspNet.SystemWeb.Tests/Castle.Facilities.AspNet.SystemWeb.Tests.csproj (hunk 1 above)
Comment (jonorossi) [reply src/Castle.Facilities.AspNet.SystemWeb.Tests/Castle.Facilities.AspNet.SystemWeb.Tests.csproj:23]: The change to the package version looks fine. We really need to move to `Directory.Packages.props`.

**Pull request merged**